run_iwyu_tests.py

```diff
@@ -111,6 +116,39 @@ def PrintLoadedTestsAndFiles():
     print('%s.%s:%s' % (cls.__name__, test, cls.test_files[test]))
 
 
+def RunTestFile(cc_file):
+  """ Executes the test for cc_file, independent of unittest infra.
+
+  Returns an appropriate exit code for the process. 0 for success, 1
+  for failure, 77 if the test was skipped. If the test failed, an
+  exception will have been raised (causing a non-zero exit code).
+  """
+  rc = 0
+  xfail = iwyu_test_util.IsTestExpectedToFail(cc_file)
+  try:
+    TestIwyuOnRelevantFiles(cc_file)
+  except unittest.SkipTest as e:
+    # Catch the skip test exception that unittest would normally handle and
+    # set an exit code that informs the test runner that test is being
+    # skipped. It's an almost-convention to use exit code 77, and some build
+    # systems natively report such failures as skipped, see e.g.
+    # https://mesonbuild.com/Unit-tests.html#skipped-tests-and-hard-errors
+    print('Skipped %s: %s' % (runner_args.run_test_file, e))
```

Comment:
runner_args.run_test_file -> cc_file here and below

Reply:
Doh! I didn't retest --run-test-file after that rebase!

Reply:
Fun thing is it actually works, see my commit message :) -- `runner_args` is global, so everything is hunky-dory.